### ISSUE TITLE
set_timeouts doesn't appear to modify timeout behavior in Firefox.  

### ISSUE THREAD

**Issue by makraiz** [stevepryde/thirtyfour_sync#3]
Using geckodriver 0.28.0 & Firefox 83.0, I tried using 'set_timeouts' to change all 3 timeouts to 180 seconds each, and the function returns an Ok result.  However, when using 'click' on a form submit button that then navigates to a new page, the operation is still timing out after 60 seconds.  I am also seeing timeouts occur at 30 seconds when using 'find_element'.
_Update: added versions to geckodriver and firefox_

**Comment by stevepryde** [comment]
Just tried this using geckodriver 0.28.0 and Firefox 83.0 on linux. I got a timeout error so the timeouts do seem to be applied.
```rust
    let caps = DesiredCapabilities::firefox();
    let driver = WebDriver::new("http://localhost:4444", &caps)?;

    let timeouts = TimeoutConfiguration::new(
        Some(Duration::from_secs(1)),
        Some(Duration::from_secs(1)),
        Some(Duration::from_secs(1)),
    );
    driver.set_timeouts(timeouts)?;
```
However, setting timeouts longer may be causing the web request to timeout instead. I can make this configurable, which will probably resolve the page load timeout issue for you. I guess you could set this to something higher than the 2nd parameter given to `TimeoutConfiguration::new`.

The `find_element` timeout will be a different setting - the 3rd parameter given to `TimeoutConfiguration::new`. That is set to 30 seconds by default, but you can set it to 0 if you prefer. Setting that to a number higher than the request timeout would give the same results as above.

I think I will make the timeout setting configurable, and will default to 120 seconds instead of 60. 180 seconds seems to high for a default in my opinion. Is there a particular reason you're wanting to set the timeout that high?

**Comment by stevepryde** [comment]
I've added `WebDriver::set_request_timeout()` to set the HTTP client timeout.
Are you able to try that out and let me know if it resolves your issue? (you'll need to point at github directly rather than crates.io)

Thanks

**Comment by makraiz** [comment]
Thanks for looking into this.  

I am working with a very slow and JavaScript bloated website, which is prone to plenty of its own errors.  The 180 second timer was set out of frustration that 'timeouts' were occurring too frequently, I agree that it is indeed too long.  I had initially tried 120 seconds, but it didn't seem to change the results.  I didn't actually time the errors until after I noticed it was still timing out frequently after I had set it to 180 seconds. 

I think you are correct in that I was mistaken in what specifically was timing out.  I tried shortening the timers to 1 second as you showed in your test and it indeed timed out correctly, and with a different error message. 

This is the error message I was getting with long timeouts in most of my cases of 'timeouts':
`The WebDriver request returned an error: error sending request for url (http://localhost:4444/session/63e7f2b3-af44-4870-b10f-6ffb0f9ea0f3/element/ae64ccce-2a56-4c12-9da1-b3f0afd81bea/click): operation timed out', src\main.rs:285:13`

The error triggered by the shortened timeout was much nicer formatted & a bit different:
 ```
Status: 500
    Additional info:
        Error: timeout
        Stacktrace:
            WebDriverError@chrome://marionette/content/error.js:181:5
            TimeoutError@chrome://marionette/content/error.js:450:5
            bail@chrome://marionette/content/sync.js:220:19
            ', src\main.rs:243:13
```

I believe you were able to determine the true cause of my problem despite my error in fully understanding the issue. 

I have redirected my cargo.toml to your GitHub to test the new function.  I did have to change my 'driver' variable to mutable to get it to compile.  I've been running tests for the majority of my work shift and have not seen a timeout of either of the above varieties since changing the request timeout.  I'm going to keep running tests, but I think this issue is completely resolved with your new set_request_timeout() method.  

Thanks again for looking into and solving this problem!
